reference-implementation/lib/writable-stream.js

```diff
@@ -173,11 +176,26 @@ function WritableStreamFinishClose(stream) {
   defaultWriterClosedPromiseResolve(stream._writer);
 }
 
-function WritableStreamFulfillWriteRequest(stream) {
-  assert(stream._writeRequests.length > 0);
+function WritableStreamRejectUnresolvedPromises(stream) {
+  const state = stream._state;
+  assert(state === 'writable' || state === 'closing' || state === 'errored');
```

Comment:
This assertion might be tightened if we move the state assignment as discussed above.